src/grappler/ngraph_optimizer.cc

```diff
@@ -65,18 +66,79 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
   // passes become a no-op.
   if (config::IsEnabled() == false ||
       std::getenv("NGRAPH_TF_DISABLE") != nullptr) {
-    NGRAPH_VLOG(0) << "Ngraph is disabled ";
+    NGRAPH_VLOG(0) << "[NGTF-OPTIMIZER] Ngraph is disabled ";
+    graph.ToGraphDef(output);
     return Status::OK();
   }
 
   // Get the nodes to be skipped
-  std::vector<string> fetch_nodes;
+  std::set<string> fetch_nodes;
   for (const string& f : item.fetch) {
-    NGRAPH_VLOG(5) << "Skip fetch node: " << f;
     int pos = f.find(":");
-    fetch_nodes.push_back(f.substr(0, pos));
+    fetch_nodes.insert(f.substr(0, pos));
+  }
+  std::set<string>& skip_these_nodes = fetch_nodes;
+
+  // Rewrite graph to add identity node so the skip node can be encapsulated
+  // as well
+  Graph* input_graph = &graph;
```

Comment:
We can move this graph rewrite into a separate function call like the other graph-rewrite function calls (MarkForClustering, AssignClusters) etc.

Reply:
So `MarkForClustering`, `AssignClusters` etc are used by both grappler path and normal path. This code is specific to grappler path.

So the options we have for the identityN adding code are:
1. the code stays here
2. the code is parceled off to a separate file, but it stays in `src/grappler`
3. The code is parceled off into a separate and placed in `src`

I  like the idea that this portion is a separate function (Then it will be easy to unit test it etc). But unsure where to place it, `src/grappler` or `src`.

Reply:
option 2. function in separate file in src/grappler

Reply:
next PR